  ‎
<h1><center>Praktikum Pemodelan Statistika terapan</center></h1>
<h2><center>Dosen Pengampu Ronny Sutsetyoko	</center></h2>
  ‎
  ‎
  ‎
  ‎

#

<img src="logo pens.png" 
        alt="Picture" 
        width="380" 
        height="380" 
        style="display: block; margin: 0 auto" />

#

#

#

#

<h4><center>Wahyu Ikbal Maulana</center></h4>

<h4><center>3323600056</center></h4>

<h4><center>D4 SDT B</center></h4>

<h2><center>Politeknik Elektronika Negeri Surabaya</center></h2>
  ‎ 


## Percobaan ke-1: Studi Kasus 1

In [1]:
library(readr)
tissue <- read_csv("breast_tissue.csv") # Checking the structure of adult data
str(tissue)

NameError: name 'library' is not defined

In [12]:
tissue <- tissue[, -1]
tissue$Class <- as.factor(tissue$Class)
levels(tissue$Class)[levels(tissue$Class)   %in%   c("fad",   "gla", "mas")] <- "other"
levels(tissue$Class)

[1] "adi"   "car"   "con"   "other"

<div style="border-radius: 10px; border: #0ea5e9 solid; padding: 15px; background-color: #ffffff00; font-size: 100%; text-align: left;">
    <b>🖊️ Analisis :</b> Dari dataset BreastTissue, kemudian direduksi kelas dimana fibro-adenoma, mastopati, dan glandular dijadikan satu kelas diberi nama ‘others’.
</div>

In [13]:
#Splitting the data using a function from dplyr package
library(caret)
library(ggplot2)
index <- createDataPartition(tissue$Class, p = .70, list = FALSE)
train <- tissue[index,]
test <- tissue[-index,]

In [14]:
# Setting the reference
train$Class <- relevel(train$Class, ref = "adi")

In [21]:
library(nnet)
# Training the multinomial model
multinom_model <- multinom(Class ~ ., data = tissue)
# Checking the model
summary(multinom_model)

# weights:  44 (30 variable)
initial  value 146.947202 
iter  10 value 107.785951
iter  20 value 71.796827
iter  30 value 17.709626
iter  40 value 11.374947
iter  50 value 6.775080
iter  60 value 5.795296
iter  70 value 5.581882
iter  80 value 4.879180
iter  90 value 4.199680
iter 100 value 4.160692
final  value 4.160692 
stopped after 100 iterations


Call:
multinom(formula = Class ~ ., data = tissue)

Coefficients:
      (Intercept)          I0      PA500        HFS         DA         Area
car      86.45689 -0.92420610  35.426343 -28.995689 -2.7614791 -0.009627356
con      65.29309 -0.02131995   3.507425   5.180119  0.3799671 -0.006557466
other    94.35136 -0.44817721 -10.573482  45.314817 -0.2506004 -0.011107358
          `A/DA`  `Max IP`         DR           P
car   -0.6522402 2.3039134  3.0645844  0.67990594
con    1.3401948 0.2354590 -0.1683327 -0.07552706
other  1.5460282 0.1120201  0.6247094  0.29486854

Std. Errors:
      (Intercept)        I0        PA500         HFS        DA        Area
car    0.03476847 0.1066177 0.0177918233 0.037347078 0.6130240 0.029524661
con    0.00246189 0.5774913 0.0003169089 0.002276567 0.3423545 0.005154037
other  0.03434195 0.1013918 0.0176815177 0.037334873 0.6274376 0.029559302
         `A/DA`  `Max IP`        DR         P
car   0.5424341 0.4562191 0.5804292 0.1709975
con   0.2034479 0.678955

<div style="border-radius: 10px; border: #0ea5e9 solid; padding: 15px; background-color: #ffffff00; font-size: 100%; text-align: left;">
    <b>🖊️ Analisis :</b> Dalam regresi logistik multinomial memakai library nnet, menghasilkan model training yang mengiterasi optimisasi train model. Optimisasi tersebut bertujual untuk meminimalkan fungsi objektivitas untuk menemukan model parameter yang optimal. Optimisasi terhenti setelah mencapai kriteria convergence, saat dimana improvisasi objek sudah kecil dan maksimum iterasi tercapai. Nilai AIC yang cenderung rendah menunjukkan bahwa model memiliki kualitas yang baik.
</div>

In [16]:
exp(coef(multinom_model))

(Intercept)  I0        PA500        HFS          DA         Area     
car   3.529816e+37 0.3968464 2.429210e+15 2.554656e-13 0.06319822 0.9904188
con   2.272107e+28 0.9789057 3.336224e+01 1.777040e+02 1.46223641 0.9934640
other 9.468380e+40 0.6387915 2.558557e-05 4.786024e+19 0.77833332 0.9889541
      `A/DA`    `Max IP`  DR         P        
car   0.5208776 10.013292 21.4255565 1.9736921
con   3.8197877  1.265489  0.8450727 0.9272546
other 4.6927944  1.118535  1.8677032 1.3429498

In [17]:
head(round(fitted(multinom_model), 2),20)

adi car  con other
1  0   0.97 0   0.03 
2  0   1.00 0   0.00 
3  0   1.00 0   0.00 
4  0   1.00 0   0.00 
5  0   1.00 0   0.00 
6  0   0.96 0   0.04 
7  0   0.77 0   0.23 
8  0   0.53 0   0.47 
9  0   1.00 0   0.00 
10 0   1.00 0   0.00 
11 0   1.00 0   0.00 
12 0   1.00 0   0.00 
13 0   0.91 0   0.09 
14 0   1.00 0   0.00 
15 0   1.00 0   0.00 
16 0   1.00 0   0.00 
17 0   1.00 0   0.00 
18 0   0.55 0   0.45 
19 0   1.00 0   0.00 
20 0   0.96 0   0.04

In [18]:
tail(round(fitted(multinom_model), 2),20)

adi car con other
87  1   0   0   0    
88  1   0   0   0    
89  1   0   0   0    
90  1   0   0   0    
91  1   0   0   0    
92  1   0   0   0    
93  1   0   0   0    
94  1   0   0   0    
95  1   0   0   0    
96  1   0   0   0    
97  1   0   0   0    
98  1   0   0   0    
99  1   0   0   0    
100 1   0   0   0    
101 1   0   0   0    
102 1   0   0   0    
103 1   0   0   0    
104 1   0   0   0    
105 1   0   0   0    
106 1   0   0   0

<div style="border-radius: 10px; border: #0ea5e9 solid; padding: 15px; background-color: #ffffff00; font-size: 100%; text-align: left;">
    <b>🖊️ Analisis :</b> Regresi multinomial memprediksi probabilitas pengamatan tertentu untuk menjadi bagian dari kelas tertentu. Kolom mewakili tingkat klasifikasi dan baris mewakili pengamatan. 20 baris pertama terklasifikasi sebagai carsinoma dan 20 baris terakhir tergolong sebagai adipose
</div>

In [19]:
# Predicting the values for train dataset
train$ClassPredicted <- predict(multinom_model, newdata = train, "class")
# Building classification table
tab <- table(train$Class, train$ClassPredicted)
# Calculating accuracy - sum of diagonal elements divided by total obs
round((sum(diag(tab))/sum(tab))*100,2)

[1] 98.68

In [20]:
# Predicting the class for test dataset
test$ClassPredicted  <-  predict(multinom_model,  newdata  =  test, "class")
# Building classification table
tab <- table(test$Class, test$ClassPredicted)
tab

       
        adi car con other
  adi     6   0   0     0
  car     0   6   0     0
  con     0   0   4     0
  other   0   0   0    14

<div style="border-radius: 10px; border: #0ea5e9 solid; padding: 15px; background-color: #ffffff00; font-size: 100%; text-align: left;">
    <b>🖊️ Analisis :</b> Didapat untuk akurasi skor dari tiap tiap konfigurasi dan semua plot yang. Didapat bahwa akurasinya sangat tinggi yaitu 98.68%. Disimpulkan bahwa model bagus dan stabil.
</div>